### ISSUE TITLE
How do I give different learning rates for each layer?

### ISSUE THREAD

**Issue by GiuseppeMinardi** [lukemelas/EfficientNet-PyTorch#205]
How do I fine-tune the network giving each layer a different learning rate? Usually in pytorch to do so is necessary to give to the optimizer a list of dictionaries such as: `torch.optim.Adam([{"param": model.network.layer1.parameters(), "lr":lr}], lr)` but given my model:

```
class TransferEfficientNet(nn.Module):
    """Feedfoward neural network with 1 hidden layer"""
    def __init__(self, classes=10):
        super().__init__()
        # Use a pretrained model
        self.network = EfficientNet.from_pretrained("efficientnet-b1")
        
        # Replace last layer
        num_ftrs = self.network._fc.in_features
        self.network._fc = nn.Sequential(nn.Linear(num_ftrs, 512), 
                                         nn.ReLU(),  
                                         nn.Dropout(0.25),
                                         
                                         nn.Linear(512, 128), 
                                         nn.ReLU(),  
                                         nn.Dropout(0.50), 
                                        
                                         nn.Linear(128,classes))
    
    def forward(self, xb):
        out = self.network(xb)
        return out

model = TransferEfficientNet()
```

I can't find a way to do so. How do I give to the optimizer different learning rates?

**Comment by lukemelas** [comment]
Yup, you can do the normal thing, but the layers have different names (i.e. it's not `model.network.layer1` but instead `model.network.somethingelse`). You can see what the names of these layers are if you print out the model. Let me know how this works.

**Comment by GiuseppeMinardi** [comment]
It worked. thank you
